modules/misc/xdg-mime-apps.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.xdg.mimeApps;
+
+  strListOrSingleton = with types; coercedTo
+    (either (listOf str) str)
+    (v: if isList v then v else [v])
+    (listOf str);
+
+  writeAssocs = let
+    writeAssoc = key: vals: "${key}=${concatStringsSep ";" vals}";
+  in assocs: concatStringsSep "\n" (mapAttrsToList writeAssoc assocs);
+
+in {
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.xdg.mimeApps = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to manage <filename>$XDG_CONFIG_HOME/mimeapps.list</filename>.
+        This option requires <option>xdg.enable</option> to be enabled.
+        </para>
+        <para>
+        The generated file is read-only.
+      '';
+    };
+
+    # descriptions from
+    # https://specifications.freedesktop.org/mime-apps-spec/mime-apps-spec-1.0.1.html
+
+    addedAssociations = mkOption {
+      type = types.attrsOf strListOrSingleton;
+      default = { };
+      example = literalExample ''
+        {
+          "mimetype1" = [ "foo1.desktop" "foo2.desktop" "foo3.desktop" ];
+          "mimetype2" = "foo4.desktop";
+        }
+      '';
+      description = ''
+        The <option>addedAssociations</option> group defines additional
+        associations of applications with mimetypes, as if the .desktop file
+        was listing this mimetype in the first place.
+      '';
+    };
+
+    removedAssociations = mkOption {
+      type = types.attrsOf strListOrSingleton;
+      default = { };
+      example = literalExample ''
+        {
+          "mimetype1" = "foo5.desktop";
+        }
+      '';
+      description = ''
+        The <option>removedAssociations</option> group removes associations of
+        applications with mimetypes, as if the .desktop file was NOT listing
+        this mimetype in the first place.
+      '';
+    };
+
+    defaultApplications = mkOption {
+      type = types.attrsOf strListOrSingleton;
+      default = { };
+      example = literalExample ''
+        {
+          "mimetype1" = [ "default1.desktop" "default2.desktop" ];
+        }
+      '';
+      description = ''
+        The <option>defaultApplications</option> group indicates the default
+        application to be used for a given mimetype. This is for instance the
+        one that will be started when double-clicking on a file in a file
+        manager. If the application is no longer installed, the next
+        application in the list is attempted, and so on.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = config.xdg.enable;
+        message = "xdg.enable is required to use xdg.mimeApps options.";
+      }
+    ];
+
+    xdg.configFile."mimeapps.list".text = ''
```

Comment:
I would suggest something like
```
xdg.configFile."mimeapps.list".text =
  let
    joinValues = mapAttrs (n: concatStringsSep ";")
  in
    generators.toINI {} {
      "Added Associations" = joinValues cfg.addedAssociations;
      "Removed Associations" = joinValues cfg.removedAssociations;
      "Default Applications" = joinValues cfg.defaultApplications;
    };
```
(untested)

Reply:
This works :+1: